# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../minised/fixes.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

AFAICS all commands are executed - a /d deletes all and a /p prints all.
I fixed this to only execute when the address matched. I wonder how it could
be released this way.

There is more stuff to fix / implement ...

  - Rene Rebe <rene@exactcode.de>

diff -u sed-1.4-orig/sedexec.c sed-1.4/sedexec.c
--- sed-1.4-orig/sedexec.c	2003-12-29 22:36:33.000000000 +0100
+++ sed-1.4/sedexec.c	2005-06-26 08:59:43.000000000 +0200
@@ -102,8 +102,10 @@
 		{
 			/* all no-address commands are selected */
 			if (ipc->addr1)
-				if (selected(ipc) == TRUE)
+				if (selected(ipc) == FALSE) {
+					ipc++;	/* not selected, next cmd */
 					continue;
+				}
 	doit:
 			command(ipc);	/* execute the command pointed at */
 
